DROD/RoomWidget.cpp

```diff
@@ -1155,6 +1170,26 @@ void CRoomWidget::HighlightSelectedTile()
 			bRemoveHighlightNextTurn = false;
 		}
 		break;
+		case T_HORN_SQUAD:
+		{
+			MovementType eMovement = GetHornMovementType(this->pCurrentGame->swordsman.GetMovementType());
```

Comment:
Though I haven't checked, it feels like there is some duplication of game engine logic here.

For improved maintainability, it may make sense to wrap this logic in a method call, possibly in CCurrentGame.  For instance, refactor the code getting goalX/Y for each of the two horns into a method in DRODLib named something like `GetNearestEntranceForHorn(wX,wY)`, and call that method here for each case.

Reply:
I was able to trim out the movement type stuff into a general function for finding horn entrances for a given entity type. It makes `CCurrentGame::BlowHorn` a little tidier, and probably makes add more horns slightly easier.